### ISSUE TITLE
Using ghc-options at the top-level in project encouraged by docs but unavailable.

### ISSUE THREAD

**Issue by philderbeast** [haskell/cabal#8781]
**Describe the bug**
The behaviour of setting `ghc-options` at the top-level in `cabal.project` (or `cabal.project.local`) doesn't agree with the docs.

> Package options affect the building of specific packages. There are three ways a package option can be specified:
>
> * They can be specified at the top-level, in which case they apply only to local package, or
> * They can be specified inside a package stanza, in which case they apply to the build of the package, whether or not it is local or external.
> * They can be specified inside an package * stanza, in which case they apply to all packages, local ones from the project and also external dependencies.
SOURCE: [Project | package-configuration-options](https://github.com/haskell/cabal/blob/master/doc/cabal-project.rst#package-configuration-options)

**To Reproduce**
For a `cabal.project`, add a `cabal.project.local` with one line `ghc-options: -Werror`

```
$ cabal build all --project-file=cabal.project
Warning:
/.../cabal.project.local:
Unrecognized field 'ghc-options' on line 1
```

**Expected behavior**
That `ghc-options` can be added to projects at the top-level to apply to local packages.

**System information**
 - Ubuntu 22.04.1 LTS

```
$ cabal --version
cabal-install version 3.8.1.0
compiled using version 3.8.1.0 of the Cabal library 
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 9.2.5
```

**Comment by ulysses4ever** [comment]
Thank you for the report. I think you need this in your cabal.project:

```
program-options
  ghc-options: -Werror
```

Documentation improvements are welcome as PRs... The part you reference does mention the program-options stanza but I agree it's confusing in the current form.

**Comment by philderbeast** [comment]
Thanks @ulysses4ever that works. So the "program" that program-options are for is GHC is it?

**Comment by ulysses4ever** [comment]
@philderbeast yes, it looks like it. But to be honest, I am not sure, and I don't think it's a good naming…

**Comment by ulysses4ever** [comment]
`cabal-options` would make so much more sense...

**Comment by philderbeast** [comment]
I've been using `program-options` for applying warnings (like `-Werror`) but I'm still seeing source paths printed when building with:

```cabal
program-options
  ghc-options: -fhide-source-paths
```

Correction, `-fhide-source-paths` does work but only after `cabal clean`.

**Comment by turion** [comment]
> Correction, `-fhide-source-paths` does work but only after `cabal clean`.

I'm observing this as well. Deleting the `program-options` block has no effect without `cabal clean`!

**Comment by Mikolaj** [comment]
Wow, that merits a new ticket (unless there is one already).
